cmd/probe/observation/main.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	"github.com/apecloud/kubeblocks/cmd/probe/internal/observation"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"go.uber.org/automaxprocs/maxprocs"
+)
+
+func main() {
+	// set GOMAXPROCS
+	_, _ = maxprocs.Set()
```

Comment:
`maxprocs.Set()` set the max processor num, it can get the exact amount of container's cpu , instead of the machine's cpus. It can avoid the scheduler to do unnecessary context switch. But I think it's ok to remove this code.